src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1293,1368 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+        this->resetReportingVariables();
+        return;
+    }
+
+    // get setpoint on the load side outlet
+    Real64 loadSideOutletSetpointTemp = this->getLoadSideOutletSetPointTemp(state);
+
+    // Use a logic similar to that for a boilder: If the specified load is 0.0 or the boiler should not run
+    // then we leave this subroutine. Before leaving
+    // if the component control is SERIESACTIVE we set the component flow to inlet flow so that flow resolver
+    // will not shut down the branch
+    auto &sim_component(DataPlant::CompData::getPlantComponent(state, this->loadSidePlantLoc));
+    bool RunFlag = true;
+    if ((this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating && currentLoad <= 0.0) || !RunFlag) {
+        if (sim_component.FlowCtrl == DataBranchAirLoopPlant::ControlType::SeriesActive)
+            this->loadSideMassFlowRate = state.dataLoopNodes->Node(this->loadSideNodes.inlet).MassFlowRate;
+        this->resetReportingVariables();
+        return;
+    }
+
+    DataPlant::PlantLoopData &thisLoadPlantLoop = state.dataPlnt->PlantLoop(this->loadSidePlantLoc.loopNum);
+    Real64 CpLoad = FluidProperties::GetSpecificHeatGlycol(state,
+                                                           thisLoadPlantLoop.FluidName,
+                                                           state.dataLoopNodes->Node(this->loadSideNodes.inlet).Temp,
+                                                           thisLoadPlantLoop.FluidIndex,
+                                                           "PLFFHPEIR::simulate()");
+
+    // Set the current load equal to the FFHP load
+    Real64 FFHPloadSideLoad = currentLoad; // this->loadSidePlantLoad = MyLoad;
+
+    if (this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) {
+
+        // Initialize the delta temperature to zero
+        Real64 FFHPDeltaTemp = 0.0; // C - FFHP inlet to outlet temperature difference, set in all necessary code paths so no initialization required
+
+        if (thisLoadPlantLoop.LoopSide(this->loadSidePlantLoc.loopSideNum).FlowLock == DataPlant::FlowLock::Unlocked) {
+            // Either set the flow to the Constant value or calculate the flow for the variable volume
+            if ((this->flowMode == DataPlant::FlowMode::Constant) || (this->flowMode == DataPlant::FlowMode::NotModulated)) {
+                // Then find the flow rate and outlet temp
+                this->loadSideMassFlowRate = this->loadSideDesignMassFlowRate;
+                PlantUtilities::SetComponentFlowRate(
+                    state, this->loadSideMassFlowRate, this->loadSideNodes.inlet, this->loadSideNodes.outlet, this->loadSidePlantLoc);
+
+                if ((this->loadSideMassFlowRate != 0.0) &&
+                    ((this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating && currentLoad > 0.0))) {
+                    FFHPDeltaTemp = currentLoad / this->loadSideMassFlowRate / CpLoad;
```

Comment:
I really hate these double divides. At least add some parentheses please.

Reply:
Addressed in earlier commits.